### PR TITLE
Service side properties

### PR DESCRIPTION
**Pull request by mvidner**
Until https://www.rubydoc.info/gems/ruby-dbus renders it, see [Reference.md](https://github.com/mvidner/ruby-dbus/pull/86/files#diff-d1e2c9575dca93a103955cef931e823c95399de444ac256b702f1faf46c70723R258) and the [API comments in DBus::Object](https://github.com/mvidner/ruby-dbus/pull/86/files#diff-e49d9b184ce0535677161317eaaba823722bbf5bac3a18cc7b6ccf444a1809f9R98)

### Example

(`rake doc:examples; cat examples/doc/exporting_properties.rb`)

```ruby
#! /usr/bin/env ruby
require "dbus"
# https://github.com/mvidner/ruby-dbus/blob/master/doc/Reference.md#exporting-properties
class Note < DBus::Object
  dbus_interface "net.vidner.Example.Properties" do
    # A read-write property "Title",
    # with `title` and `title=` accessing @title.
    dbus_attr_accessor :title, DBus::Type::STRING
    # A read-only property "Author"
    # (type specified via DBus signature)
    # with `author` reading `@author`
    dbus_attr_reader :author, "s"
    # A read-only property `Clock`
    def clock
      Time.now.to_s
    end
    dbus_reader :clock, "s"
    # Name mapping: `CreationTime`
    def creation_time
      "1993-01-01 00:00:00 +0100"
    end
    dbus_reader :creation_time, "s"
    dbus_attr_accessor :book_volume, DBus::Type::VARIANT, dbus_name: "Volume"
  end
  dbus_interface "net.vidner.Example.Audio" do
    dbus_attr_accessor :speaker_volume, DBus::Type::BYTE, dbus_name: "Volume"
  end
  # Must assign values because `nil` would crash our connection
  def initialize(opath)
    super
    @title = "Ahem"
    @author = "Martin"
    @book_volume = 1
    @speaker_volume = 11
  end
end
obj = Note.new("/net/vidner/Example/Properties")
bus = DBus::SessionBus.instance
service = bus.request_service("net.vidner.Example")
service.export(obj)
main = DBus::Main.new
main << bus
main.run
```

### Tests

- [x] pass locally, `rake spec`
- [x] pass with `rake osc:build` on Tumbleweed (local problems with non-matching abuild user? edit /etc/passwd and chown home!)
- [x] reintroduce CI on GItHub instead of the discontinued Travis-CI.org: https://github.com/mvidner/ruby-dbus/pull/93

### Documentation

- [x] Review the `doc/` files to update them regarding properties
- [x] make the `example/` work

### CODE REVIEW

File: spec/service_newapi.rb [hunk 1]
```diff
@@ -119,6 +119,140 @@ def initialize(path)
     dbus_signal :LongTaskEnd
   end
 
+=begin
```
Comment (mvidner) [inline spec/service_newapi.rb:122]: This file has leftover notes and ideas. As long as this comment is here, ignore this file.

**Comment by lslezak** [comment]
Thanks, LGTM so far.

File: lib/dbus/introspect.rb [hunk 2]
```diff
@@ -210,4 +223,39 @@ def to_xml
...
+    # @return [String] The name of the property, for example FooBar.
+    attr_reader :name
+    attr_reader :type
+    # @return [Symbol] :read :write or :readwrite
+    attr_reader :access
+
+    # @return [Symbol] What to call at Ruby side. (TODO clarify :foo vs :foo=)
+    attr_reader :ruby_name
+
+    def initialize(name, type, access, ruby_name:)
```
Comment (jreidinger) [inline lib/dbus/introspect.rb:239]: well, when you use named param, why not have default `ruby_name: name.to_sym`? so
```suggestion
def initialize(name, type, access, ruby_name: name.to_sym)
```

File: spec/main_loop_spec.rb [hunk 3]
```diff
@@ -70,7 +70,7 @@ def test_loop_quit(delay)
     @obj.on_signal "LongTaskEnd"
   end
 
-  it "tests loop quit" do
+  it "tests loop quit", slow: true do
```
Comment (jreidinger) [inline spec/main_loop_spec.rb:73]: :+1:

File: lib/dbus/introspect.rb (hunk 2 above)
Comment (mvidner) [reply lib/dbus/introspect.rb:239]: No, a simple `to_sym` conversion is not wanted and I do want to keep the logic out of this class.

Having missed it for methods and signals, for properties I use the correct naming conventions: `foo_bar` at the Ruby side is `FooBar` at the D-Bus side.

**Comment by mvidner** [comment]
Coverage check fails because the new CI misses the code covered by the asynchronously started *service* and counts only the code run directly by RSpec. Locally it works :-/

**Pull request merged**